app/controllers/provider/admin/cms/switches_controller.rb

```diff
@@ -24,7 +24,7 @@ def destroy
 
   def find_switch
     # https://github.com/3scale/system/issues/3162
-    @switch = current_account.hideable_switches.stringify_keys[params[:id]]
+    @switch = current_account.hideable_switches[params[:id]]
```

Comment:
There is a bug here 😞 And the problem is this doesn't have a test yet so it didn't catch it, but imagine like if in `test/integration/admin/cms/switches_controller_test.rb` we had this (I've done it locally to check 😉 ):
```ruby
# frozen_string_literal: true

require 'test_helper'

class Provider::Admin::CMS::SwitchesControllerTest < ActionDispatch::IntegrationTest
  def setup
    @provider = FactoryBot.create(:provider_account)
    login! provider
    plan = ApplicationPlan.new(issuer: master_account.first_service!, name: 'enterprise')
    provider.force_upgrade_to_provider_plan!(plan)
  end

  attr_reader :provider

  test 'Finance is globally disabled' do
    ThreeScale.config.stubs(onpremises: false)
    get provider_admin_cms_switches_path
    assert_response :success
    assert_select '#switch-finance-toggle'

    ThreeScale.config.stubs(onpremises: true)
    get provider_admin_cms_switches_path
    assert_response :success
    assert_select '#switch-finance-toggle', true
    assert_select %(table#switches th), text: 'Finance', count: 1
  end

  test 'update shows the hidden switch' do
    switch_name = provider.hideable_switches.keys.first
    put provider_admin_cms_switch_path(switch_name, format: :js)
    assert_response :success
    assert provider.settings.switches[switch_name].visible?
  end
end
```
If you run this in the `master` branch, it works, because `params[:id]` is a String, and it stringify keys. But if you run that test in this branch, it fails (raises `ActiveRecord::RecordNotFound` and returns `404`) because it tries to find from an String (`params[:id]`) among `Symbols` (https://github.com/3scale/porta/pull/697/files#diff-3e724f642f3565db93f8553d54794326R128 & https://github.com/3scale/porta/pull/697/files#diff-3e724f642f3565db93f8553d54794326R6) so it can not find it.

And we cannot do `params[:id].to_sym` for what it says in the issue of the previous line. And btw, imo we shouldn't point to an issue of an old private repository when it is not necessary because that info is public in other pages, for example in https://codeclimate.com/changelog/5172ca1489af7e7abe0071f9/ : 
![image](https://user-images.githubusercontent.com/11318903/54123653-8d21c480-4400-11e9-8560-e66f89883bf5.png)

So a possible way to make it work is changing this line https://github.com/3scale/porta/pull/697/files#diff-3e724f642f3565db93f8553d54794326R128 from:
```ruby
delegate *%i[[] each keys values fetch partition map include? any? all? empty? first with_indifferent_access delete], to: :switches
```
to:
```ruby
delegate *%i[[] stringify_keys each keys values fetch partition map include? any? all? empty? first with_indifferent_access delete], to: :switches
```
And in this line putting back the `.stringify_keys` as we had before.
But to be honest I don't find it a very clean solution... I would prefer having there a method `find!` that does everything that this method `find_switch` is doing here 😄 that way this controller doesn't need to know about how the switches are implemented inside.

Reply:
I guess we weren't supposed to remove `stringify_keys` from here. I'll put it back. If it's red without it, that's probably because of bad test. We should fix the tests instead.

Reply:
> If it's red without it, that's probably because of bad test

More than bad test, lack of test 😂 That's why I wrote it myself in the example I gave you 😄